packages/iobio-react-components/src/utils.ts

```diff
@@ -52,3 +62,68 @@ export const setElementStyles = (element: Element, styles: string) => {
 		element.shadowRoot.adoptedStyleSheets = [elementStyles];
 	}
 };
+
+/**
+ * Obtain BAM statistical data from Data Broker data events
+ * @param dataEvent { [BamKey]: number  }
+ */
+
+type StatisticsData = { [K in StatisticKey]: number };
+type PercentData = { [K in BamPercentKey]: number };
+
+type DataUpdate = StatisticsData & PercentData;
+
+export const getBamStatistics = (dataEvent: DataUpdate) => {
+	return [...percentKeys, ...statisticKeys].reduce(
+		(statsData, dataKey) => {
+			const value = dataEvent[dataKey];
+			const stats: { [k: string]: number } = { ...statsData, [dataKey]: value };
+
+			if (percentKeys.some((percentKey) => percentKey === dataKey)) {
+				const percentage = Number((value / dataEvent['total_reads']).toPrecision(4));
+				const key = `${dataKey}_percentage`;
+				stats[key] = percentage;
+			}
+			return stats;
+		},
+		{} as { [k: string]: number },
+	);
+};
+
+/**
+ * Obtain BAM Histogram data from Data Broker data events
+ * @param dataEvent { [K in BamHistogramKey]: { [numKey: string]: number } }
+ * where numKey parses to an Integer
+ */
+
+type HistogramData = { [K in BamHistogramKey]: { [numKey: string]: number } };
+
+export const getHistogramData = (dataEvent: HistogramData) => {
+	const histogramData = histogramKeys.reduce((acc, val) => {
```

Comment:
missed the `acc, val` declarative renaming here

Reply:
Updated